music_assistant/providers/plex_connect/timeline.py

```diff
@@ -0,0 +1,468 @@
+        self.mass.subscribe(self._handle_queue_time_update, EventType.QUEUE_TIME_UPDATED)
+        self.mass.subscribe(self._handle_player_update, EventType.PLAYER_UPDATED)
+        self.logger.info("Plex timeline reporter setup complete - listening for events")
+
+    async def close(self) -> None:
+        """
+        Clean up resources and close connections.
+
+        Terminates all running tasks and closes the HTTP session.
+        """
+        self.logger.debug("Closing Plex timeline reporter...")
+        for task in self._tasks.values():
+            if not task.done():
+                task.cancel()
+                with contextlib.suppress(asyncio.CancelledError):
+                    await task
+
+        if self._session:
+            await self._session.close()
+            self._session = None
+        self.logger.info("Plex timeline reporter closed")
+
+    async def _handle_queue_update(self, event) -> None:
+        queue_id = event.object_id
+        queue = event.data
+        if not queue or not queue_id:
+            return
+
+        self.logger.debug("Queue update for %s, state: %s", queue_id, queue.state)
+
+        current_item = queue.current_item
+        if not current_item or not current_item.streamdetails:
+            self.logger.debug("Queue %s has no current item or streamdetails", queue_id)
+            self._stop_reporting(queue_id)
+            return
+
+        if not self._is_plex_item(current_item):
+            self.logger.debug("Current item is not from Plex provider")
+            self._stop_reporting(queue_id)
+            return
+
+        plex_data = current_item.streamdetails.data
+        if not self._validate_plex_data(plex_data):
+            self.logger.debug("Missing required Plex data in streamdetails")
+            return
+
+        if queue.state == PlayerState.PLAYING:
+            self._handle_playing_state(queue_id, queue, current_item, plex_data)
+        elif queue.state == PlayerState.PAUSED:
+            self._handle_paused_state(queue_id, queue, current_item)
+        elif queue.state == PlayerState.IDLE:
+            await self._handle_idle_state(queue_id, queue, current_item)
+
+    def _is_plex_item(self, current_item: QueueItem) -> bool:
+        if not current_item.streamdetails:
+            return False
+        provider = current_item.streamdetails.provider
+        if not provider or not provider.startswith(("plex:", "plex--")):
+            return False
+        return isinstance(current_item.streamdetails.data, dict)
+
+    def _validate_plex_data(self, plex_data: dict) -> bool:
+        required_keys = ["rating_key", "server_url", "token", "machine_identifier"]
+        if not all(k in plex_data for k in required_keys):
+            missing = ", ".join(k for k in required_keys if k not in plex_data)
+            self.logger.debug("Missing required Plex data in streamdetails: %s", missing)
+            return False
+        return True
```

Comment:
So this seems to hard depend on the plex provider itself to send this data ?
Maybe then it makes sense to add "depends_on" in the manifest and set it to plex